clang/lib/Format/UnwrappedLineFormatter.cpp

```diff
@@ -616,6 +627,62 @@ class LineJoiner {
     return 1;
   }
 
+  unsigned tryMergeNamespace(SmallVectorImpl<AnnotatedLine *>::const_iterator I,
+                             SmallVectorImpl<AnnotatedLine *>::const_iterator E,
+                             unsigned Limit) {
+    if (Limit == 0)
+      return 0;
+    if (I[1]->InPPDirective != (*I)->InPPDirective ||
+        (I[1]->InPPDirective && I[1]->First->HasUnescapedNewline)) {
+      return 0;
+    }
+    if (I + 2 == E || I[2]->Type == LT_Invalid)
+      return 0;
+
+    Limit = limitConsideringMacros(I + 1, E, Limit);
+
+    if (!nextTwoLinesFitInto(I, Limit))
+      return 0;
+
+    // Check if it's a namespace inside a namespace, and call recursively if so.
+    // '3' is the sizes of the whitespace and closing brace for " _inner_ }".
+    if (I[1]->First->is(tok::kw_namespace)) {
+      if (I[1]->Last->is(tok::comment))
```

Comment:
```suggestion
      if (I[1]->Last->is(tok::comment) || !Style.CompactNamespaces)
```
And adjust the test cases and/or turning `CompactNamespaces` on/off. See https://github.com/llvm/llvm-project/pull/105597#issuecomment-2408847951.